    /**
     * Tests IO exception, that can occur during reading of properties file.
     */
    @Test
    public void testIoException() throws Exception {
        final DefaultConfiguration checkConfig = createModuleConfig(OrderedPropertiesCheck.class);
        final OrderedPropertiesCheck check = new OrderedPropertiesCheck();
        check.configure(checkConfig);
        final String fileName =
                getPath("InputOrderedPropertiesCheckNotExisting.properties");
        final File file = new File(fileName);
        final FileText fileText = new FileText(file, Collections.emptyList());
        final SortedSet<Violation> violations =
                check.process(file, fileText);
        assertEquals(1, violations.size(), "Wrong violations count: " + violations.size());
        final Violation violation = violations.iterator().next();
        final String retrievedMessage = violations.iterator().next().getKey();
        assertEquals("unable.open.cause", retrievedMessage,
                "violation key '" + retrievedMessage + "' is not valid");
        assertEquals(violation.getViolation(),
                getCheckMessage(MSG_IO_EXCEPTION_KEY, fileName, getFileNotFoundDetail(file)),
                "violation '" + violation.getViolation() + "' is not valid");
    }

